src/apps/lwaftr/fragmentv4.lua

```diff
@@ -0,0 +1,100 @@
+module(..., package.seeall)
+
+local constants = require("apps.lwaftr.constants")
+local packet = require("core.packet")
+local ffi = require("ffi")
+
+local band, bor, lshift = bit.band, bit.bor, bit.lshift
+local ntohs, htons = ffi.C.ntohs, ffi.C.htons
+
+local int16_ptr_t = ffi.typeof("uint16_t*")
+
+local total_length_offset = constants.ethernet_header_size + constants.o_ipv4_total_length
+local frag_id_offset = constants.ethernet_header_size + constants.o_ipv4_identification
+local flags_and_frag_offset_offset = constants.ethernet_header_size + constants.o_ipv4_flags
+local ihl_and_ver_offset = constants.ethernet_header_size + constants.o_ipv4_ver_and_ihl
+
+-- Constants to manipulate the flags next to the frag-offset field directly
+-- as a 16-bit integer, without needing to shift the 3 flag bits.
+local flag_dont_fragment_mask  = 0x4000
+local flag_more_fragments_mask = 0x2000
+local frag_offset_field_mask   = 0x1FFF
+
+
+local function get_u16_at_offset(pkt, offset)
+	return ntohs(ffi.cast(int16_ptr_t, pkt.data + offset)[0])
+end
+
+local function set_u16_at_offset(pkt, offset, value)
+	ffi.cast(int16_ptr_t, pkt.data + offset)[0] = htons(value)
+end
+
```

Comment:
At some point I'd love to see a function like this generalized because this an idiom that it's used quite often in all the codebase and I think having it wrapped within a function will help to communicate better what the code does. We trust LuaJIT will end up inlining it.

I'm OK with the name, but since this operation is repeated often perhaps we could use a shorter name like write16 or read16 (for 32-bit, write32 and read32), or maybe even one single function that do both things:

```
local function word16(pkt, offset, value)
   if not value then
      return ffi.cast(int16_ptr_t, pkt.data + offset)[0])
   end
   ffi.cast(int16_ptr_t, pkt.data + offset)[0] = value
end
```

For the moment, leave it like this.